src/figuresClasses.ts

```diff
@@ -1,11 +1,99 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
you can create function like

```
function checkOnError(...args: number[]): void {
  if (args.some((arg: number) => arg <= 0)) {
    throw new Error('Some side less 0');
  }
}
```
It is needed because you have the same logic in three places.